modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,375 @@
+			 * images to avoid making individual database calls.
+			 */
+			_prime_post_caches( $attachment_ids, false, true );
+		}
+
+		// Iterate through the matches in order of occurrence as it is relevant for whether or not to lazy-load.
+		foreach ( $matches as $match ) {
+			// Filter an image match.
+			if ( isset( $images[ $match[0] ] ) ) {
+				$filtered_image = $match[0];
+				$attachment_id  = $images[ $match[0] ];
+				/**
+				 * Filters img tag that will be injected into the content.
+				 *
+				 * @param string $filtered_image the img tag with attributes being created that will
+				 *                                    replace the source img tag in the content.
+				 * @param string $context Optional. Additional context to pass to the filters.
+				 *                        Defaults to `current_filter()` when not set.
+				 * @param int $attachment_id the ID of the image attachment.
+				 *
+				 * @since 1.0.0
+				 */
+				$filtered_image = apply_filters( 'dominant_color_img_tag_add_adjust', $filtered_image, $context, $attachment_id );
+
+				if ( $filtered_image !== $match[0] ) {
+					$content = str_replace( $match[0], $filtered_image, $content );
+				}
+			}
+		}
+
+		return $content;
+	}
+
+	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+	foreach ( $filters as $filter ) {
+		add_filter( $filter, 'dominant_color_filter_content_tags', 20 );
+	}
+
+	add_filter( 'dominant_color_img_tag_add_adjust', 'dominant_color_tag_add_adjust', 10, 3 );
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominantcolor]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @return string[]
+ */
+function dominant_color_set_image_editors() {
+
+	require_once 'class-dominant-color-image-editor-gd.php';
+	require_once 'class-dominant-color-image-editor-imagick.php';
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @param integer $id the image id.
+ * @param string  $default_color default color.
+ *
+ * @return string
+ * @since n.e.x.t
+ */
+function dominant_color_get( $id, $default_color = 'eeeeee' ) {
+	/**
+	 * Filters the default color to use when no dominant color is found.
+	 *
+	 * @param string $default_color The default color.
+	 *
+	 * @since n.e.x.t
+	 */
+	$default_color = apply_filters( 'dominant_color_default_color', $default_color );
```

Comment:
I like the idea of allowing to set a default, _however_ this default should not be applied in here. Let me clarify why: This function is used to permanently set metadata in the attachment. A default should be used for when a real value is not set, but we shouldn't use a default to actually set the value to it. That makes it impossible for us to know in the future whether that value was actually set intentionally or whether it's just a default.

Let's remove the default handling from this function and instead apply it only when using the dominant color information to modify the image in the frontend. This way we don't pollute metadata with a default and keep ourselves open to improving e.g. the detection in the future.

I think it would be better for this function to return a `WP_Error` in case anything fails.

Reply:
I have made this change